ansys/mapdl/core/mapdl.py

```diff
@@ -2225,19 +2315,38 @@ def load_table(self, name, array, var1="", var2="", var3=""):
         >>> mapdl.parameters['MY_TABLE']
         array([0.0001, 0.0001, 0.0005, 0.0005, 0.0002, 0.0002])
         """
-        if array.ndim < 2:
+        if not isinstance(array, np.ndarray):
+            raise ValueError("The table should be a Numpy array")
+        if array.shape[0] < 2 or array.shape[1] < 2:
+            raise ValueError("One or two of the array dimensions are too small to create a table.")
+        if array.ndim == 2:
+            self.dim(name, "TABLE", imax=array.shape[0]-1, jmax=array.shape[1]-1, kmax="",
+                     var1=var1, var2=var2, var3=var3, csysid=csysid)
+        elif array.ndim == 3:
+            self.dim(name, "TABLE", imax=array.shape[0]-1, jmax=array.shape[1]-1, kmax=array.shape[2],
+                     var1=var1, var2=var2, var3=var3, csysid=csysid)
+        else:
             raise ValueError(
-                "Expecting at least a 2D table, but input contains " "only 1 dimension"
+                f"Expecting only a 2D or 3D table, but input contains\n{array.ndim} dimensions"
             )
-        self.dim(name, "TABLE", array.shape[0], var1=var1, var2=var2, var3=var3)
+
+        if any(array[0, :] != np.sort(array[0, :])):
+            raise ValueError('The underlying ``TREAD`` command requires that the axis 0 is in ascendent order.')
+        if any(array[:, 0] != np.sort(array[:, 0])):
+            raise ValueError('The underlying ``TREAD`` command requires that the axis 1 is in ascendent order.')
```

Comment:
Spelling error, and also sorting is slow when checking if an array is in ascending order. Instead see https://stackoverflow.com/a/47004507

```suggestion
        if not np.all(array[0, :-1] <= array[0, 1:]):
            raise ValueError('The underlying ``TREAD`` command requires that the axis 0 is in ascending order.')
        if not np.all(array[:-1, 0] <= array[1:, 0]):
            raise ValueError('The underlying ``TREAD`` command requires that the axis 1 is in ascending order.')
```

Reply:
Mmmhh.. very nice suggestion. Thank you!